internal/js/modules/k6/browser/common/frame_session.go

```diff
@@ -1045,6 +1045,8 @@ func (fs *FrameSession) attachWorkerToTarget(ti *target.Info, session *Session)
 		return fmt.Errorf("attaching worker target ID %v to session ID %v: %w",
 			ti.TargetID, session.ID(), err)
 	}
+	fs.page.workersMu.Lock()
+	defer fs.page.workersMu.Unlock()
```

Comment:
In the browser module, we put these locks into types that are responsible for them. In this case, it's worth creating a new method (e.g., `Page.addWorker`) for this locking logic under `Page`.

Reply:
Resolved in [4bd0de1](https://github.com/grafana/k6/pull/4716/commits/4bd0de1fe708162923272b8d436ca7c0aca0ccf0)